### PR TITLE
Changing Timesketch behavior: using the API client and importer client.

### PR DESCRIPTION
**Pull request by kiddinn**
This PR changes the behavior in dftimewolf to start using the Timesketch API and importer clients instead of implementing their own API client.

This changes also makes use of latest changes in the API client to store configuration settings to save user input.

### CODE REVIEW

**Comment by kiddinn** [comment]
tests run locally on my machine

File: dftimewolf/lib/exporters/timesketch.py [hunk 1]
```diff
@@ -22,86 +27,130 @@ class TimesketchExporter(module.BaseModule):
...
+    if not sketch:
+      sketch = self.timesketch_api.get_sketch(self.sketch_id)
 
-    sketch_url = self.timesketch_api.GetSketchUrl(self.sketch_id)
+    recipe_name = self.state.recipe.get('name', 'no_recipe')
+    input_names = []
+    for description, _ in self.state.input:
+      if not description:
+        continue
+      name, _, _ = description.rpartition('.')
```
Comment (tomchop) [inline dftimewolf/lib/exporters/timesketch.py:117]: isn't `name = description.rpartition('.')[0]` a bit clearer?

File: dftimewolf/lib/exporters/timesketch.py [hunk 2]
```diff
@@ -22,86 +27,130 @@ class TimesketchExporter(module.BaseModule):
...
+      name = name.replace(' ', '_')
+      name = name.replace('-', '_')
```
Comment (tomchop) [inline dftimewolf/lib/exporters/timesketch.py:119]: You can chain these

File: dftimewolf/lib/exporters/timesketch.py [hunk 3]
```diff
@@ -22,86 +27,130 @@ class TimesketchExporter(module.BaseModule):
...
+
+    with importer.ImportStreamer() as streamer:
+      streamer.set_sketch(sketch)
+      streamer.set_timeline_name(timeline_name)
+
+      for _, path in self.state.input:
+        streamer.add_file(path)
+
+    api_root = sketch.api.api_root
+    host_url, _, _ = api_root.partition('api/v1')
```
Comment (tomchop) [inline dftimewolf/lib/exporters/timesketch.py:136]: Same, can you use `[0]` here?

File: dftimewolf/lib/exporters/timesketch.py (hunk 1 above)
Comment (kiddinn) [reply dftimewolf/lib/exporters/timesketch.py:117]: I'm not sure I agree with it being clearer, but nonetheless no strong feelings here so done....

File: dftimewolf/lib/exporters/timesketch.py (hunk 2 above)
Comment (kiddinn) [reply dftimewolf/lib/exporters/timesketch.py:119]: yes... just a question which one is more readable... chained

File: dftimewolf/lib/exporters/timesketch.py (hunk 3 above)
Comment (kiddinn) [reply dftimewolf/lib/exporters/timesketch.py:136]: done

**Comment by kiddinn** [comment]
done with this round, PTAL @tomchop

**Pull request merged**